apps/dashboard/src/components/primitives/copy-button.tsx

```diff
@@ -1,20 +1,77 @@
-import { HoverToCopy } from '@/components/primitives/hover-to-copy';
-import { cn } from '@/utils/ui';
-import { RiFileCopyLine } from 'react-icons/ri';
+import { AnimatePresence, motion } from 'motion/react';
+import { useState } from 'react';
+import { RiCheckLine, RiFileCopyLine } from 'react-icons/ri';
+import { cn } from '../../utils/ui';
 import { Button, ButtonProps } from './button';
+import { Tooltip, TooltipContent, TooltipTrigger } from './tooltip';
 
 type CopyButtonProps = ButtonProps & {
   valueToCopy: string;
+  inputGroup?: boolean;
 };
 
 export const CopyButton = (props: CopyButtonProps) => {
-  const { className, valueToCopy, children, ...rest } = props;
+  const { className, valueToCopy, inputGroup, children, ...rest } = props;
+
+  const [copied, setCopied] = useState<boolean>(false);
+
+  const handleCopy = async () => {
+    try {
+      await navigator.clipboard.writeText(valueToCopy);
+      setCopied(true);
+      setTimeout(() => setCopied(false), 1500);
+    } catch (err) {
+      console.error('Failed to copy text: ', err);
+    }
+  };
+
+  const sizeClass = props.size === '2xs' ? 'size-3' : 'size-4';
 
   return (
-    <HoverToCopy asChild valueToCopy={valueToCopy}>
-      <Button mode="outline" variant="secondary" className={cn('flex items-center gap-1', className)} {...rest}>
-        {children || <RiFileCopyLine className="size-4" />}
-      </Button>
-    </HoverToCopy>
+    <Tooltip>
+      <TooltipTrigger asChild>
+        <Button
+          mode="outline"
+          variant="secondary"
+          className={cn(className, inputGroup && 'h-[34px] rounded-none border-l ring-0')}
+          onClick={handleCopy}
+          aria-label={copied ? 'Copied' : 'Copy to clipboard'}
+          disabled={copied}
+          {...rest}
+        >
+          {children}
+          <div className={`relative ${sizeClass}`}>
+            <AnimatePresence mode="wait" initial={false}>
```

Comment:
Based on the preview deployment the timing between hidding the icon and showing the green check should be reduced a little bit.